src/gitpod.ts

```diff
@@ -0,0 +1,236 @@
+ * Configure where in the IDE the terminal should be opened.
+ */
+export enum GitpodOpenIn {
+  /**
+   * the bottom panel (default)
+   */
+  BOTTOM = 'bottom',
+  /**
+   * the left panel
+   */
+  LEFT = 'left',
+  /**
+   * the right panel
+   */
+  RIGHT = 'right',
+  /**
+   * the main editor area
+   */
+  MAIN = 'main'
+}
+
+/**
+ * Configure options for a task to be run when opening a Gitpod
+ * workspace (e.g. running tests, or starting a dev server).
+ *
+ * Start Mode	        | Execution
+ * Fresh Workspace	  | before && init && command
+ * Restart Workspace	| before && command
+ * Snapshot	          | before && command
+ * Prebuild	          | before && init && prebuild
+*/
+export interface GitpodTask {
+
+  /**
+   * Required. What to run
+   */
+  readonly command: string;
+
+  /**
+   * A name for this
+   * @default
+   */
+  readonly name?: string;
+
+  /**
+   * You can configure how the terminal should be opened relative to the previous task.
+   * @default
+   */
+  readonly openMode?: GitpodOpenMode;
+
+  /**
+   * You can configure where in the IDE the terminal should be opened
+   * @default BOTTOM
+   */
+  readonly openIn?: GitpodOpenIn;
+
+  /**
+   * In case you need to run something even before init, that is a requirement for both init and command,
+   * you can use the before property.
+   * @default
+   */
+  readonly before?: string;
+
+  /**
+   *
+   * The init property can be used to specify shell commands that should only be executed after a workspace
+   * was freshly cloned and needs to be initialized somehow. Such tasks are usually builds or downloading
+   * dependencies. Anything you only want to do once but not when you restart a workspace or start a snapshot.
+   * XXX: NodeProject
+   * @default "yarn install"
+   */
+  readonly init?: string;
+
+  /**
+   * The optional prebuild command will be executed during prebuilds. It is meant to run additional long running
+   * processes that could be useful, e.g. running test suites.
+   * @default
+   */
+  readonly prebuild?: string;
+}
+
+/**
+ * What can we configure for the GitPod component
+ */
+export interface GitpodOptions {
+  /**
+   * Optional Docker Configuration
+   * Defaults to https://github.com/gitpod-io/workspace-images/blob/master/full/Dockerfile
+   * @default undefined
+   */
+  readonly docker?: GitpodDocker;
+
+  /**
+   * This must be defaulted per project
+   * @default []
+   */
+  readonly tasks: GitpodTask[];
+}
+
+/**
+ * The Gitpod component itself
+ */
+export class Gitpod extends Component {
+  /**
+   * The List of tasks. The default will be present or a customization
+   */
+  private tasks = new Array<GitpodTask>();
+
+  /**
+   * We default to jsii/superchain fo the image, so this can be required
+   */
+  private docker: GitpodDocker;
+
+  /**
+   *
+   * @param project The project
+   * @param options The component configuration options for this project
+   */
+  constructor(project: Project, options?: GitpodOptions) {
```

Comment:
I think it's fine to omit the docstring for the constructor here.
```suggestion
  constructor(project: Project, options?: GitpodOptions) {
```